Add onboarding & release info to OneLocBuild doc (#7374)
dotnet/arcade: Documentation/OneLocBuild.md

To make OneLocBuild easier to use, we have integrated the task into Arcade. This integration is a job template
([here](/eng/common/templates/job/onelocbuild.yml)) that is described in this document.

## Onboarding to OneLocBuild Using Arcade

Onboarding to OneLocBuild is a simple process:

1. Ensure that your repository is on the latest version of Arcade.
2. Create a test branch (e.g. `LocalizationTests`) in your repository and add the following job template to your YAML:
```yaml
- template: /eng/common/templates/job/onelocbuild.yml
  parameters:
    CreatePr: false
```
3. Run the pipeline you want to use OneLocBuild on your test branch.
4. Open a ticket with the localization team using
   [this template](https://ceapex.visualstudio.com/CEINTL/_workitems/create/Loc%20Request). Include the link to the 
   test build you've done.

> 5. The loc team will generate an LCL package for you and send you its name. It will be something like

Reviewer: ```suggestion
5. The loc team will generate an LCL package for you and send you its ID. It will be something like
```

I suggest "ID" instead of "name" to match the phrase "package ID" in the next step.

   `LCL-JUNO-PROD-YOURREPO`.
6. Change your YAML (subbing `'LCL-JUNO-PROD-YOURREPO'` with the package ID given to you) to:
```yaml
- template: /eng/common/templates/job/onelocbuild.yml
  parameters:
    LclSource: lclFilesfromPackage
    LclPackageId: 'LCL-JUNO-PROD-YOURREPO'
```

> 7. Merge the changes to your main branch and then let open a

Reviewer: ```suggestion
7. Merge the changes to your main branch and then open a
```

   [repo modification ticket](https://ceapex.visualstudio.com/CEINTL/_workitems/create/Loc%20Request) with the loc team 
   to let them know to retarget the branch.

> *Note: as of 12 May 2021, if your repository is mirrored to internal with Maestro, you will also need to keep*

Reviewer: > if your repository is mirrored to internal with Maestro

Would you say that "most" repos that will read this are mirrored this way?

Reviewer: If so, consider making it the default example and have an explanation about exceptions.

Author: I think I want it to be clear that this is a temporary thing and not a permanent state.

> *Note: as of 12 May 2021, if your repository is mirrored to internal with Maestro, you will also need to keep*

Reviewer: > Note: as of 12 May 2021...

I suggest making this whole section a normal part of the documentation. This is the way things are. If it changes in the future, we update the documentation.

*`CreatePr: false` in your YAML. Currently, OneLocBuild does not support our scenario, so PRs are being made*

> *manually before release. Please notify Jon Fortescue (jofortes) if you're following this step so that you can*

Reviewer: > so PRs are being made manually before release

This is awkward for me to read. Is this the limitation that you're trying to work-around (because of mirroring, PRs are created before release and we don't want that) or is this the work-around (Because of mirroring, PRs are made manually and before release)?

Author: it's the workaround. I will see if I can clarify this.

Reviewer: As long as it doesn't make the whole thing too overloaded, call out the steps in the "normal" process that fail in our environment, exactly is done for a work-around, and how that work-around actually works in terms that make sense for localization.

Author: how does current draft look?

Reviewer: Looks great!

> *manually before release. Please notify Jon Fortescue (jofortes) if you're following this step so that you can*

Reviewer: > Please notify Jon Fortescue (jofortes)

Consider making this a dnceng team contact. Maybe even FR or simply point [here](https://github.com/dotnet/core-eng/wiki/How-to-get-a-hold-of-Engineering-Servicing).

*be updated when PR creation is supported.*

## Releasing with OneLocBuild Using Arcade

> **NB: The SLA for translations is one week. Please allow at least two weeks from the release for this process.**

Reviewer: What means "NB"?

Author: yeah i'll change this to "note" lol

it stands for "nota bene"

Reviewer: 🌈

### If You're Releasing from `main`
If you're releasing from the main branch of your repository, all that you need to do is ensure that you're merging
PRs from OneLocBuild as they are made and that you allow the translator SLA for any new strings prior to the release.

### If You're Releasing from a Branch Other Than `main` (Including Servicing Branches)
If you're releasing from any other branch (including servicing branches), you must do the following:

> 1. Add the OneLocBuild task to the pipeline YAML of the release branch

Reviewer: > OneLocBuild task

Probably should be "job template"

2. Open a [repo modification ticket](https://ceapex.visualstudio.com/CEINTL/_workitems/create/Loc%20Request) with the 

>    loc team at least two weeks before the release and tell them to re-target your repository to the release branch.

Reviewer: ```suggestion
   loc team at least two weeks before the release and request that they re-target your repository to the release branch.
```

4. Merge the OneLocBuild PRs to your release branch.

> 5. After the release, tell Cristiano to re-target your repository to the `main` branch again.

Reviewer: >  tell Cristiano to re-target

Is it correct to call out Cristiano specifically? Or is there a better general path?

Author: no, this is a remnant from a previous draft. thank you for the catch!

Reviewer 2: Please open a ticket for re-targeting as well.
